src/verbosity.jl

```diff
@@ -0,0 +1,89 @@
+    # used to fix the logging source file + line
+    # since we're lowering our verbose logging macros to the
+    # Logging.jl macros; otherwise, they would always report this (verbosity.jl)
+    # file as the logging callsite
+    expr.args[1].args[2] = src
+    return expr
+end
+
+vlogmacrodocs = """
+    @debugv N msg args...
+    @infov N msg args...
+    @warnv N msg args...
+    @errorv N msg args...
+
+"Verbose" logging macros. Drop in replacements of standard logging macros, but an
+additional verbosity level `N` is passed to indicate differing verbosity levels
+for a given log level. The verbosity argument is subtracted from the base log level when passed down
+to the core logging logic, so `@debugv 1 msg` will essentially call `@logmsg Debug-1 msg`.
+
+An `LoggingExtras.LevelOverrideLogger`can then be used to filter on the `level` argument.
+For convenience, the
+[`LoggintExtras.with(f; level, verbosity)`](@ref) function is provided to temporarily
+wrap the current logger with a log level and verbosity subtracted to filter while `f` is executed.
+"""
+
+macro debugv(verbosity::Int, msg, exs...)
+    return restore_callsite_source_position!(
+        esc(:($Base.@logmsg (Logging.Debug - $verbosity) $msg $(exs...))),
+        __source__,
+    )
+end
+
+macro infov(verbosity::Int, msg, exs...)
+    return restore_callsite_source_position!(
+        esc(:($Base.@logmsg (Logging.Info - $verbosity) $msg $(exs...))),
+        __source__,
+    )
+end
+
+macro warnv(verbosity::Int, msg, exs...)
+    return restore_callsite_source_position!(
+        esc(:($Base.@logmsg (Logging.Warn - $verbosity) $msg $(exs...))),
+        __source__,
+    )
+end
+
+macro errorv(verbosity::Int, msg, exs...)
+    return restore_callsite_source_position!(
+        esc(:($Base.@logmsg (Logging.Error - $verbosity) $msg $(exs...))),
+        __source__,
+    )
+end
+
+macro logmsgv(verbosity::Int, level, msg, exs...)
+    return restore_callsite_source_position!(
+        esc(:($Base.@logmsg ($level - $verbosity) $msg $(exs...))),
+        __source__,
+    )
+end
+
+@eval @doc $vlogmacrodocs :(@logmsgv)
+@eval @doc $vlogmacrodocs :(@debugv)
+@eval @doc $vlogmacrodocs :(@infov)
+@eval @doc $vlogmacrodocs :(@warnv)
+@eval @doc $vlogmacrodocs :(@errorv)
+
+"""
+    LoggingExtras.with(f; level=Info)
+
+Convenience function like `Logging.with_logger` to temporarily wrap
+the current logger with a level filter while `f` is executed.
+That is, the current logger will still be used for actual logging, but
+log messages will first be checked that they meet the `level`
+log level before being passed on to be logged.
+"""
+function with(f; level::Union{Int, LogLevel}=Info, verbosity::Integer=0)
+    lvl = Base.CoreLogging._min_enabled_level[]
+    try
+        # by default, this global filter is Debug, but for debug logging
+        # we want to enable sub-Debug levels
+        Base.CoreLogging._min_enabled_level[] = BelowMinLevel
```

Comment:
I kind of want to do this in the `__init__` of LoggingExtras.
And then open an issue on JuliaLang/julia and cross reference it in the `__init__`
One thing I am not sure on though is if it should be `BelowMinLevel` or `BelowMinLevel+1`?

Reply:
I thought about that as well (whether we should just do it in `__init__`), but I worried about the case where someone is actually _using_ this functionality (I don't think it's publicly documented, but still), and then if they load LoggingExtras, all the sudden it resets the `_min_enabled_level` to the lowest level.

So this is just me trying to be ultra-conservative and make sure it doesn't interact with anything anybody might be already doing with this, IMO unnecessary and scary, global.

Reply:
Doesn't block this PR though